tests/CustomListTest.php

```diff
@@ -150,4 +150,44 @@ protected function getListType(): string
 
         $this->assertSame(2, count($stringList));
     }
+
+    /** @test */
+    public function it_should_concatenate_another_custom_list()
+    {
+        $items = ['item', 'anotherItem'];
+
+        $stringList = new class ($items) extends CustomList {
+            protected function getListType(): string
+            {
+                return 'string';
+            }
+        };
+
+        $otherItems = ['otherItem', 'anotherOtherItem'];
+
+        $otherStringList = new class ($otherItems) extends CustomList {
+            protected function getListType(): string
+            {
+                return 'string';
+            }
+        };
+
+        $otherItems2 = ['otherItem2', 'anotherOtherItem2'];
+
+        $otherStringList2 = new class ($otherItems2) extends CustomList {
+            protected function getListType(): string
+            {
+                return 'string';
+            }
+        };
+
+        $stringList->merge($otherStringList, $otherStringList2);
+
+        $this->assertSame('item', $stringList[0]);
+        $this->assertSame('anotherItem', $stringList[1]);
+        $this->assertSame('otherItem', $stringList[2]);
+        $this->assertSame('anotherOtherItem', $stringList[3]);
+        $this->assertSame('otherItem2', $stringList[4]);
+        $this->assertSame('anotherOtherItem2', $stringList[5]);
+    }
```

Comment:
I would add one more test checking that the function throws an exception if elements are not the same type when merging two or more lists.